src/main/java/uk/ac/ebi/eva/evaseqcol/controller/seqcol/SeqColController.java

```diff
@@ -28,25 +29,36 @@ public SeqColController(SeqColService seqColService) {
     }
 
     @GetMapping(value = "/{digest}")
-    public ResponseEntity<SeqColEntity> getSeqColByDigestAndLevel(
+    public ResponseEntity<?> getSeqColByDigestAndLevel(
             @PathVariable String digest, @RequestParam(required = false) String level) {
-        if (level == null) {
-            level = "none";
+        if (!level.equals("1") && !level.equals("2")) {
```

Comment:
Still need to check that `level` is non-null here...

Reply:
True, 
Actually the order of the conditions was not correct.
Just fixed it.

Reply:
> Or actually, can you make this the `default` case in your switch statement? Then it happens after the null check anyway.

`case "1"` and `case null` have actually the same code, but we can't put `case null` in Java (at least for our version), so I think adding the `default` case will lead to code duplicate..

Reply:
Ah okay got you,
I think your approach looks more beautiful , so I'll go for it